### ISSUE TITLE
ElasticsearchRest Client not working with scheduler

### ISSUE THREAD

**Issue by jstuczyn** [CogStack/CogStack-Pipeline#31]
While the first batch of jobs is going to run fine, the next one is going to throw the following exception:
```
java.lang.IllegalStateException: Request cannot be executed; I/O reactor status: STOPPED
```

From what I've managed to gather, it is probably due to the way Elasticsearch RestClient is using Apache's HttpAsyncClient. For some reason after the batch the client is closed? It might require further investigation.

The possible solution might include recreating RestClient for each batch (however, that would be rather inefficient) or somehow trying to determine if the client is still alive.

For now, the Native Java client can be used (I have updated it to work with X-pack). However, Elastic recommends the migration to the Rest client in a future.

**Comment by afolarin** [comment]
OK let's take a look at this tomorrow. AFAIK closing after the batch doesn't sound like the right behaviour

**Comment by afolarin** [comment]
cc/ @HeglerTissot

**Comment by lrog** [comment]
A provisional fix for the scheduler has been implemented in 56080d798caf383713b9ce9280d5e5ea4a8fedd8 .

The general problem is that the ES REST Client (implemented in `ESRestService`) needs to be closed at the end of batch processing, otherwise the services and resources deinitialization process gets blocked. The `@PreDestroy` annotated method in ESRestService is not called automatically hence not releasing the resource.

The current workaround in the code (in `JobCompleteNotificationListener::afterJob()`) closes the ES REST client at the end of running a job and it works when running single jobs (`SingleJobLauncher`). However, when running scheduled jobs, the jobs are continuously executed, calling each time `JobCompleteNotificationListener::afterJob()`. Hence, when trying to write to ElasticSearch using ES REST Client for second time it will fail as the client has been already deinitialized (hence the exception mentioned by @jstuczyn ).

To overcome this issue, a simple fix was implemented -- re-initializing the ES REST client in `JobCompleteNotificationListener::beforeJob()` method, checking whether the client has been previously deinitialised.

However, this is just a quick fix, as ideally a slight refactoring of handling ES REST client and resources should be implemented.

**Comment by afolarin** [comment]
Note that we should revisit this issue for the refactor milestone.
